### PR TITLE
Add 'u' specifier to std.format

### PR DESCRIPTION
**Pull request by data-man**
Outputs integer as an UTF-8 sequence.

### CODE REVIEW

File: lib/std/fmt.zig [hunk 1]
```diff
@@ -585,6 +592,18 @@ pub fn formatAsciiChar(
     return output(context, @as(*const [1]u8, &c)[0..]);
 }
 
+pub fn formatUtf8Codepoint(
+    c: u32,
```
Comment (daurnimator) [inline lib/std/fmt.zig]: Should be a `u21`.

Comment (data-man) [reply lib/std/fmt.zig]: Why?
std.unicode's funcs doesn't use this restriction.

Comment (daurnimator) [reply lib/std/fmt.zig]: > Why?

Its the actual acceptable range of a utf8 encoder.

>  std.unicode's funcs doesn't use this restriction.

Ah, it probably should.

Comment (data-man) [reply lib/std/fmt.zig]: Also, a codepoint may be in a struct's u32 field, so uglifed cast will be needed.

Comment (daurnimator) [reply lib/std/fmt.zig]: > Ah, it probably should.

Done in https://github.com/ziglang/zig/pull/3987

File: lib/std/fmt.zig [hunk 2]
```diff
@@ -587,6 +594,18 @@ pub fn formatAsciiChar(
...
 
+pub fn formatUtf8Codepoint(
+    c: u21,
+    options: FormatOptions,
+    context: var,
+    comptime Errors: type,
+    output: fn (@TypeOf(context), []const u8) Errors!void,
+) Errors!void {
+    var buf: [4]u8 = undefined;
+    const len = std.unicode.utf8Encode(c, buf[0..]) catch unreachable;
```
Comment (daurnimator) [inline lib/std/fmt.zig]: why is this catch unreachable? what if someone e.g. passed in `0x1fffff`

File: lib/std/fmt.zig [hunk 3]
```diff
@@ -520,6 +521,12 @@ pub fn formatIntValue(
         } else {
             @compileError("Cannot print integer that is larger than 8 bits as a ascii");
         }
+    } else if (comptime std.mem.eql(u8, fmt, "u")) {
+        if (@TypeOf(int_value).bit_count <= 21) {
```
Comment (daurnimator) [inline lib/std/fmt.zig]: Does a comptime int pass this check?

File: lib/std/fmt.zig (hunk 2 above)
Comment (data-man) [reply lib/std/fmt.zig]: > attempt to unwrap error: CodepointTooLarge

Personally I'd prefer to have `utf8EncodeUnsafe` with 0xFFFD replacement.

File: lib/std/fmt.zig (hunk 3 above)
Comment (data-man) [reply lib/std/fmt.zig]: Copy-pasted from `c` modifier.

**Comment by data-man** [comment]
Works for me as is. You got an idea.

**Pull request closed without merging**